pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,427 @@
+                axis.linkedViewChanged)
+            axis_view = axis.linkedView()
+            if cvb is not axis_view:
+                # FROM ViewBox.linkView
+                # connext axis's view changes to view since axis acts just like a proxy to it
+                if axis.orientation in ["top", "bottom"]:
+                    # connect axis main view changes to view
+                    cvb.state["linkedViews"][cvb.XAxis] = weakref.ref(axis_view)
+                    # this signal is received multiple times when using mouse actions directly on the viewbox
+                    # this causes the non top layer views to scroll more than the frontmost one
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigXRangeChanged"] = axis_view.sigXRangeChanged.connect(
+                        cvb.linkedXChanged)
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigResized"] = axis_view.sigResized.connect(
+                        cvb.linkedXChanged)
+                    # disable autorange on manual movements
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigXRangeChangedManually"] = axis_view.sigXRangeChangedManually.connect(
+                        lambda mask: self.disableAxisAutoRange(axis_name))
+                elif axis.orientation in ["right", "left"]:
+                    # connect axis main view changes to view
+                    cvb.state["linkedViews"][cvb.YAxis] = weakref.ref(axis_view)
+                    # this signal is received multiple times when using mouse actions directly on the viewbox
+                    # this causes the non top layer views to scroll more than the frontmost one
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigYRangeChanged"] = axis_view.sigYRangeChanged.connect(
+                        cvb.linkedYChanged)
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigResized"] = axis_view.sigResized.connect(
+                        cvb.linkedYChanged)
+                    # disable autorange on manual movements
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigYRangeChangedManually"] = axis_view.sigYRangeChangedManually.connect(
+                        lambda mask: self.disableAxisAutoRange(axis_name))
+            self._signalConnectionsByChart[chart.name]["cvb.sigStateChanged"] = cvb.sigStateChanged.emit(
+                cvb)
+        # resize plotitem according to the master one
+        # resizing it's view doesn't work for some reason
+        self._signalConnectionsByChart[chart.name]["self.vb.sigResized"] = self.vb.sigResized.connect(
+            lambda vb: chart.plotItem.setGeometry(vb.sceneBoundingRect()))
+        # fix prepareForPaint by outofculture
+        self._signalConnectionsByChart[chart.name]["scene.sigPrepareForPaint"] = scene.sigPrepareForPaint.connect(
+            cvb.prepareForPaint)
+        if cvb is not tvb:
```

Comment:
to shorten the lines here I think we can do something like...

```python
signal_dictionary = self._signalConnectionsByChart[chart.name]
signal_dictionary["tvb.sigMouseDragged"] = ....
```

Reply:
used "sd" to shorten even more